.github/workflows/build.yaml

```diff
@@ -109,12 +108,23 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Run GoReleaser
+      - name: Build and release
         uses: goreleaser/goreleaser-action@v5
+        if: startsWith(github.event.ref, 'refs/tags/')
         with:
           distribution: goreleaser
           version: latest
-          args: release --rm-dist
+          args: release --clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build
+        uses: goreleaser/goreleaser-action@v5
+        if: ${{ !startsWith(github.event.ref, 'refs/tags/') }}
+        with:
+          distribution: goreleaser
+          version: latest
+          args: build --snapshot
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Just a thought:  rather than having two completely separate steps, each conditionalized on `github.event.ref`, can you instead use a single, common step with a parameter value (e.g., `${{ env.goreleaser_args }}`) whose value is set conditionally, instead?

Reply:
Found [it](https://docs.github.com/en/actions/learn-github-actions/expressions#example)!

Try this:
```
      - name: Build or release
        uses: goreleaser/goreleaser-action@v5
        if: 
        with:
          distribution: goreleaser
          version: latest
          args: ${{ env.ARGS }}
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          ARGS: $${{ startsWith(github.event.ref, 'refs/tags/') && 'release --clean' || 'build --snapshot' }}
```

Reply:
You're solution suffered the same problem mine did. The raw string `'release --clean'` is not getting interpolated by the shell, so `goreleaser` gets the command `'release --clean'` which does not exist. 

For me, adding another function to split the raw string of `ARGS`, is not worth the effort for only two cases.